frontend/src/App.tsx

```diff
@@ -1,43 +1,84 @@
 import React from 'react';
-import Supersocial from './Supersocial';
 import { Login } from './Login';
 import { Route, Switch, Redirect, BrowserRouter } from 'react-router-dom';
 import { EventBus, EventBusEventType } from './EventBus';
 import { LoginManager } from './LoginManager';
 import { Registration } from './Registration';
-
-export interface LoginInfo {
-  loggedIn?: boolean
-  username?: string
-}
+import { Sidebar } from './Sidebar';
+import { MessageOverview } from './MessageOverview';
+import { ImageProvider } from './ImageProvider';
+import { AnalyticsPage } from './AnalyticsPage';
+import { SettingsPage } from './SettingsPage';
+import { AboutPage } from './AboutPage';
+import { LandingPage } from './LandingPage';
 
 /** Main entry point for the application. The EventBus and the LoginManager live on this layer and are passed downwards from here. */
-class App extends React.Component<LoginInfo, LoginInfo> {
-  constructor(props) {
+class App extends React.Component<any> {
+  eventBus = new EventBus();
+  loginManager = new LoginManager(this.eventBus);
+
+  constructor(props: any) {
     super(props);
-    this.state = {
-      loggedIn: false,
-      username: "Not logged in"
-    };
+    this.eventBus.register(EventBusEventType.USER_CHANGE, (eventType, eventData?) => this.onUserChange(eventData));
   }
 
   private onUserChange(eventData?: any) {
     this.setState(eventData);
   }
 
+
+
+  /**Change the selected page. */
+  private selectComponent(components, selected: number) {
+    return (<div className="App">
+      <Sidebar components={components} loginManager={this.loginManager} selected={selected} />
+    </div>);
+  }
+
   render() {
-    let eventBus = new EventBus();
-    let loginManager = new LoginManager(eventBus);
-    eventBus.register(EventBusEventType.USER_CHANGE, (eventType, eventData?) => this.onUserChange(eventData));
 
+    const components = [{
```

Comment:
To simplify the check underneath, you could add a property like `protected` or `secure` or `checkLogin` for each component.